js/h264bsd_decoder.js

```diff
@@ -143,38 +143,43 @@ H264bsdDecoder.prototype.decode = function() {
     if(pInput == 0) return H264bsdDecoder.NO_INPUT;
 
     var pBytesRead = module._malloc(4);
-
-    var bytesRead = 0;
-    var retCode = module._h264bsdDecode(pStorage, pInput + inputOffset, inputLength - inputOffset, 0, pBytesRead);
-    
-    if (retCode == H264bsdDecoder.RDY ||
-        retCode == H264bsdDecoder.PIC_RDY ||
-        retCode == H264bsdDecoder.HDRS_RDY) {
-        bytesRead = module.getValue(pBytesRead, 'i32');
+    while (pInput > 0)
```

Comment:
This was critical right here. The calls in queueInput to move memory around are expensive.